modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/DLPreviewLimitConfigurationProviderImpl.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.document.library.internal.configuration;
+
+import com.liferay.document.library.configuration.DLPreviewLimitConfigurationProvider;
+import com.liferay.document.library.internal.configuration.admin.service.DLPreviewLimitManagedServiceFactory;
+import com.liferay.portal.configuration.metatype.annotations.ExtendedObjectClassDefinition;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Marco Galluzzi
+ */
+@Component(service = DLPreviewLimitConfigurationProvider.class)
+public class DLPreviewLimitConfigurationProviderImpl
+	implements DLPreviewLimitConfigurationProvider {
+
+	@Override
+	public long getMaxFileSize(String scope, long scopePK) {
+		if (scope.equals(
```

Comment:
It should be more efficient to convert `scope` to an enum value and compare with `==`.

Reply:
I've done that. However, it seems that `ExtendedObjectClassDefinition.Scope` is not intended to be used as Enum but as a string, so I had to add a method to it (similar to `valueOf`).